src/api/v1/db/oracledb/pizzas-dao.js

```diff
@@ -0,0 +1,162 @@
+ * table `resourceTableName` into an array of strings written as
+ * `TABLENAME.COLUMNAME = "RESOURCENAME_attributeName"
+ *
+ * @param {string} resourceName
+ * @param {string} resourceTableName
+ * @param {object} resourceColumns
+ * @returns {object[]}
+ */
+const resourceColumnAliases = (
+  resourceName,
+  resourceTableName,
+  resourceColumns,
+) => _.map(resourceColumns,
+  (columnName, attributeName) => `${resourceTableName}.${columnName} AS "${resourceName}_${attributeName}"`);
+
+/**
+ * Returns a list of column aliases for pizzas, and, optionally,
+ * doughs and ingredients based on the values in `included`
+ *
+ * @param {string[]} included
+ * @returns {string[]}
+ */
+const getColumnAliases = (included) => {
+  const columns = resourceColumnAliases('PIZZA', 'PIZZAS', pizzaColumns);
+
+  if (included.includes('dough')) {
+    columns.push(
+      ...resourceColumnAliases('DOUGH', 'DOUGHS', doughColumnNames),
+    );
+  }
+  if (included.includes('ingredients')) {
+    columns.push(
+      ...resourceColumnAliases('INGREDIENT', 'INGREDIENTS', ingredientsColumnNames),
+    );
+  }
+
+  return columns;
+};
+
+const innerJoinDoughs = 'LEFT JOIN DOUGHS ON PIZZAS.DOUGH_ID = DOUGHS.ID';
+
+const innerJoinIngredients = `LEFT JOIN PIZZA_INGREDIENTS ON PIZZAS.ID = PIZZA_INGREDIENTS.PIZZA_ID
+  LEFT JOIN INGREDIENTS ON INGREDIENTS.ID = PIZZA_INGREDIENTS.INGREDIENT_ID`;
+/**
+ * A query to fetch a pizza with, optionally, its ingredients and doughs
+ *
+ * @param {Array} included
+ * @returns {string}
+ */
+const getPizzaByIdQuery = (included) => `SELECT ${getColumnAliases(included).join(', ')}
+  FROM PIZZAS 
+  ${included.includes('dough') ? innerJoinDoughs : ''}
+  ${included.includes('ingredients') ? innerJoinIngredients : ''}
+  WHERE PIZZAS.ID = :id`;
+
+/**
+ * Extracts resource attributes from a row object with
+ * column names formatted like RESOURCENAME_attributeName
+ * for a resource with name `resourceName` and attributes `resourceAttributes`
+ *
+ * @param {string} resourceName should be written in all caps
+ * @param {object} resourceAttributes
+ * @param {object} row
+ * @returns {object}
+ */
+const extractRawResource = (resourceName, resourceAttributes, row) => {
+  if (row[`${resourceName}_id`] === null) return null;
+  const rawResource = {};
+  _.keys(resourceAttributes).forEach(
+    (attributeName) => {
+      rawResource[attributeName] = row[`${resourceName}_${attributeName}`];
+    },
+  );
+  return rawResource;
+};
+
+/**
+ * Turn raw database output from `fullPizzaQuery` into
+ * an array of raw dough objects with doughs and ingredients
+ *
+ * @param {Array} rows
+ * @returns {Array} an array of raw pizza objects
+ */
+const normalizePizzaRows = (rows) => {
+  let doughsIncluded = false;
+  let head;
+  let ingredientsIncluded = false;
+  let index = 0;
+  const pizzas = [];
+
+  if (rows[0]) {
+    doughsIncluded = 'DOUGH_id' in rows[0];
+    ingredientsIncluded = 'INGREDIENT_id' in rows[0];
+  }
```

Comment:
@tsoliangwu0130 follow up to our conversation earlier, here's what this looks like without the `included` parameter